designer/client/src/Designer.tsx

```diff
@@ -38,32 +34,36 @@ export class Designer extends Component<Props, State> {
     return this.props.previewUrl
   }
 
-  incrementFlyoutCounter = (callback = () => {}) => {
+  incrementFlyoutCounter = () => {
     let currentCount = this.state.flyoutCount
-    this.setState({ flyoutCount: ++currentCount }, callback())
+    this.setState({ flyoutCount: ++currentCount })
   }
 
-  decrementFlyoutCounter = (callback = () => {}) => {
+  decrementFlyoutCounter = () => {
     let currentCount = this.state.flyoutCount
-    this.setState({ flyoutCount: --currentCount }, callback())
+    this.setState({ flyoutCount: --currentCount })
   }
 
-  save = async (toUpdate, callback = () => {}) => {
-    await this.designerApi.save(this.id, toUpdate)
-    this.setState(
-      { data: toUpdate }, // optimistic save
-      callback()
-    )
-    return toUpdate
+  get = async () => {
+    const definition = await form.get(this.id)
+
+    this.setState({
+      data: definition
+    })
+
+    return definition
   }
 
-  updatePageContext = (page) => {
-    this.setState({ page })
+  save = async (definition: FormDefinition) => {
+    await form.save(this.id, definition)
+    return this.get()
```

Comment:
The save response returns the `{ ok: true }` pattern rather than the definition

It also causes `response.json()` parsing errors